drv/stm32xx-sys/src/main.rs

```diff
@@ -88,8 +90,72 @@ fn main() -> ! {
+                    // loop iteration, isn't it? Too bad, these registers all
+                    // have totally distinct types because of the PAC.
+                    match register {
+                        0 => syscfg.exticr1.modify(|_, w| match slot {
+                            // Safety: field modeled incorrectly in PAC
+                            0 => unsafe { w.exti0().bits(port as u8) },
+                            // Safety: field modeled incorrectly in PAC
+                            1 => unsafe { w.exti1().bits(port as u8) },
+                            // Safety: field modeled incorrectly in PAC
+                            2 => unsafe { w.exti2().bits(port as u8) },
+                            // Safety: field modeled incorrectly in PAC
+                            _ => unsafe { w.exti3().bits(port as u8) },
+                        }),
+                        1 => syscfg.exticr2.modify(|_, w| match slot {
+                            // Safety: field modeled incorrectly in PAC
+                            0 => unsafe { w.exti4().bits(port as u8) },
+                            // Safety: field modeled incorrectly in PAC
+                            1 => unsafe { w.exti5().bits(port as u8) },
+                            // Safety: field modeled incorrectly in PAC
+                            2 => unsafe { w.exti6().bits(port as u8) },
+                            // Safety: field modeled incorrectly in PAC
+                            _ => unsafe { w.exti7().bits(port as u8) },
+                        }),
+                        2 => syscfg.exticr3.modify(|_, w| match slot {
+                            // Safety: field modeled incorrectly in PAC
+                            0 => unsafe { w.exti8().bits(port as u8) },
+                            // Safety: field modeled incorrectly in PAC
+                            1 => unsafe { w.exti9().bits(port as u8) },
+                            // Safety: field modeled incorrectly in PAC
+                            2 => unsafe { w.exti10().bits(port as u8) },
+                            // Safety: field modeled incorrectly in PAC
+                            _ => unsafe { w.exti11().bits(port as u8) },
+                        }),
+                        _ => syscfg.exticr4.modify(|_, w| match slot {
+                            // Safety: field modeled incorrectly in PAC
+                            0 => unsafe { w.exti12().bits(port as u8) },
+                            // Safety: field modeled incorrectly in PAC
+                            1 => unsafe { w.exti13().bits(port as u8) },
+                            // Safety: field modeled incorrectly in PAC
+                            2 => unsafe { w.exti14().bits(port as u8) },
+                            // Safety: field modeled incorrectly in PAC
+                            _ => unsafe { w.exti15().bits(port as u8) },
+                        }),
+                    }
```

Comment:
this whole block of code really makes me feel very positively about the PAC :/

Reply:
YOU AND ME BOTH

Reply:
Fortunately if we want to fix how the fields are modeled, all we have to do is construct a YAML file describing a diff we want to apply to another YAML file derived from an XML file that isn't checked into the repo.